src/components/Editor/Footer.js

```diff
@@ -199,17 +198,17 @@ class SourceFooter extends PureComponent<Props> {
   }
 }
 
-export default connect(
-  state => {
-    const selectedSource = getSelectedSource(state);
-    const selectedId = selectedSource.get("id");
-    const source = selectedSource.toJS();
-    return {
-      selectedSource,
-      mappedSource: getGeneratedSource(state, source),
-      prettySource: getPrettySource(state, selectedId),
-      endPanelCollapsed: getPaneCollapse(state, "end")
-    };
-  },
-  dispatch => bindActionCreators(actions, dispatch)
-)(SourceFooter);
+const mapStateToProps = state => {
+  const selectedSource = getSelectedSource(state);
+  const selectedId = selectedSource.get("id");
+  const source = selectedSource.toJS();
+
+  return {
+    selectedSource,
+    mappedSource: getGeneratedSource(state, source),
+    prettySource: getPrettySource(state, selectedId),
+    endPanelCollapsed: getPaneCollapse(state, "end")
+  };
+};
+
+export default connect(mapStateToProps, actions)(SourceFooter);
```

Comment:
You seem to jump around between passing in `actions` and a `mapDispatchToProps` object. I kind of prefer the latter as it then limits the number of action creators passed to the component (making it easier to read prop lists in react devtools 😄 )

Reply:
I agree with the notion, the actions should either be extracted or passed as a whole - having it done a different way every time creates confusion. That is the scope for a different pr however.

The aim here was to clean up the connect, get rid of bindActionCreators etc.